lib/web_ui/lib/src/engine/view_embedder/dom_manager.dart

```diff
@@ -47,38 +137,58 @@ class DomManager {
   /// The tag name for the scene host.
   static const String sceneHostTagName = 'flt-scene-host';
 
+  /// The tag name for the text editing host.
+  static const String textEditingHostTagName = 'flt-text-editing-host';
+
   /// The tag name for the semantics host.
   static const String semanticsHostTagName = 'flt-semantics-host';
 
   /// The tag name for the accessibility announcements host.
   static const String announcementsHostTagName = 'flt-announcement-host';
 
-  final FlutterViewEmbedder _embedder;
-
   /// The root DOM element for the entire Flutter View.
   ///
   /// This is where input events are captured, such as pointer events.
   ///
   /// If semantics is enabled, this element also contains the semantics DOM tree,
   /// which captures semantics input events.
-  DomElement get rootElement => _embedder.flutterViewElementDEPRECATED;
+  DomElement rootElement;
 
   /// Hosts all platform view elements.
-  DomElement get platformViewsHost => _embedder.glassPaneElementDEPRECATED;
+  DomElement platformViewsHost;
 
   /// Hosts all rendering elements and canvases.
-  DomShadowRoot get renderingHost => _embedder.glassPaneShadowDEPRECATED;
+  DomShadowRoot renderingHost;
+
+  /// Hosts the <flt-scene> element.
+  ///
+  /// This element is created and inserted in the HTML DOM once. It is never
+  /// removed or moved. However the <flt-scene> inside of it may be replaced.
+  DomElement sceneHost;
 
   /// Hosts all text editing elements.
-  DomElement get textEditingHost => _embedder.textEditingHostNodeDEPRECATED;
+  DomElement textEditingHost;
 
   /// Hosts the semantics tree.
   ///
   /// This element is in front of the [renderingHost] and [platformViewsHost].
   /// Otherwise, the phone will disable focusing by touch, only by tabbing
   /// around the UI.
-  DomElement get semanticsHost => _embedder.semanticsHostElementDEPRECATED;
+  DomElement semanticsHost;
 
   /// This is where accessibility announcements are inserted.
-  DomElement get announcementsHost => _embedder.announcementsHostDEPRECATED;
+  DomElement announcementsHost;
+}
+
+DomShadowRoot _attachShadowRoot(DomElement element) {
+  if (getJsProperty<Object?>(element, 'attachShadow') == null) {
+    throw UnsupportedError('ShadowDOM is not supported in this browser.');
+  }
+
+  return element.attachShadow(<String, dynamic>{
+    'mode': 'open',
+    // This needs to stay false to prevent issues like this:
+    // - https://github.com/flutter/flutter/issues/85759
+    'delegatesFocus': false,
```

Comment:
I think we don't need this either, since @htoor3 moved all focusables (text input, semantics) outside the shadow root.

Reply:
I would prefer to not make functional changes in this PR :)

I'll file an issue though, so we make this change in the future.

Reply:
https://github.com/flutter/flutter/issues/138576